### PR TITLE
Update the push-fedora-40 script so it properly creates one image index

### PR DESCRIPTION
**Pull request by ipanova**
with child manifests.

[noissue]

* adds proper artifactType
* adds proper artifact layer type
* child per-acrhes manifests are tagged

### CODE REVIEW

File: push-fedora-40.sh [hunk 1]
```diff
@@ -3,28 +3,38 @@
...
+    quay.io/foreman/nboci-files:$OSVER-arm64 ./work/*
+
+rm -rf ./work/
+# push image index containing netboot artifacts for different arches
+podman manifest push --all --rm quay.io/foreman/nboci-files:$OSVER
+
+# tag child manifests
+# going to use skopeo for now https://github.com/containers/podman/issues/23599
+# Parse index image, find correspodning child manifests and tag them accordingly
+skopeo inspect docker://ghcr.io/ipanova/fedora/fedora-netboot:$OSVER --raw |jq -r '.manifests.[] | "\(.platform.architecture) \(.digest)"' | while read i; do ARCH="$(echo $i | awk '{print $1}')"; DIGEST="$(echo $i | awk '{print $2}')" ; skopeo copy docker://ghcr.io/ipanova/fedora/fedora-netboot@${DIGEST} docker://ghcr.io/ipanova/fedora/fedora-netboot:$OSVER-${ARCH} ; done
```
Comment (ekohl) [inline push-fedora-40.sh]: Does it make sense to start introducing variables for `ghcr.io/ipanova` or further?

File: push-fedora-40.sh [hunk 2]
```diff
@@ -3,28 +3,38 @@
...
+    --artifact-layer-type application/x-netboot-file \
+    quay.io/foreman/nboci-files:$OSVER-arm64 ./work/*
+
+rm -rf ./work/
+# push image index containing netboot artifacts for different arches
+podman manifest push --all --rm quay.io/foreman/nboci-files:$OSVER
+
+# tag child manifests
+# going to use skopeo for now https://github.com/containers/podman/issues/23599
+# Parse index image, find correspodning child manifests and tag them accordingly
```
Comment (ekohl) [inline push-fedora-40.sh]:
```suggestion
# Parse index image, find corresponding child manifests and tag them accordingly
```

File: push-fedora-40.sh [hunk 3]
```diff
@@ -3,28 +3,38 @@
 OS=fedora
 OSVER=fedora-40
 
-podman manifest create quay.io/foreman/nboci-files:$OSVER
+# create image index and set the annotation at the index level
+# TODO set the artifactType at the index level https://github.com/containers/podman/issues/23598
+podman manifest create quay.io/foreman/nboci-files:$OSVER --annotation org.pulpproject.netboot.version=1
 
+mkdir -p work
 rm -f ./work/*
```
Comment (ekohl) [inline push-fedora-40.sh:11]: Would it make more sense to just delete the whole directory and then recreate it? That also removes any hidden files:
```suggestion
rm -rf ./work
mkdir work
```

And perhaps out of scope, but should each step that creates/needs files create their own subdirectory in `work`? Perhaps even with `mktemp -d -p work`. That way you only do a `mkdir -p "$WORKDIR"` at the start and and the end `rm -rf "$WORKDIR"`.

File: push-fedora-40.sh (hunk 1 above)
Comment (ipanova) [reply push-fedora-40.sh]: ops sorry, i should have updated it to the foreman repo

File: push-fedora-40.sh (hunk 3 above)
Comment (lzap) [reply push-fedora-40.sh:11]: In that case please remove the work directory from git, there is I think `.gitkeep` hidden file. I like the idea of creating a temporary directory and removing it afterwards.

Comment (ekohl) [reply push-fedora-40.sh:11]: I can probably come up with a PR so should we merge this and do it as a follow up?

Comment (lzap) [reply push-fedora-40.sh:11]: Sure, let’s merge Ina just returned from longer leave and she might be busy. I am ready to do it as well if needed. This was just quick and dirty way of doing stuff, let’s fix that.

Comment (ekohl) [reply push-fedora-40.sh:11]: https://github.com/theforeman/nboci-files/pull/2 is what I came up with. It escalated a bit, but individual commits should be reasonably easy to follow.

**Pull request merged**